README.md

```diff
@@ -379,6 +379,34 @@ Succeeded   : 50
 
 </details>
 
+## Design
+
+### High-level Decisions
+
+- **Nextflow Workflow**
+  - Many QC tests will require the output from one or more commands using external (_i.e._ non-Python) programs. While we can run external commands in Python (_e.g. using the `subprocess` module), we want to avoid re-inventing a workflow management system (WMS). Instead, we should leverage an existing WMS. For the purposes of this POC, we opted for Nextflow since we have the compute infrastructure set up. That said, a future iteration could use the Common Workflow Language (whose reference implementation is in Python) and/or the Task Execution Service (TES).
+  - Given that Nextflow supports arbitrary Groovy/Java code, we could technically accomplish all of the work in Nextflow alone. However, it makes a lot more sense to use a programming language that our team (_i.e._ DPE) is familiar with and that people outside of our team can readily contribute to. Python fits the bill. Hence, we are currently building a Python-Nextflow hybrid tool. 
+  - In Nextflow, the `container` directive and the `script` block can leverage values provided as inputs (_e.g._ `container "${some_input}"`). This functionality allows us to keep the Nextflow workflow very general. Otherwise, we would have to hard-code the containers and commands in the Nextflow workflow and somehow keep everything in sync with the Python package, which takes care of interpreting the commands.
+  - The Nextflow workflow makes use of a short `parseJson()` Groovy function to parse the contents of a JSON file into a Nextflow channel. This helps bridge the gap between Python-based processes and Nextflow since Nextflow cannot easily handle multiple outputs that are not files (_e.g._ stdout), especially when the output is intended to be structured.
+- **Python Package**
+  - I use the term "QC target" to refer to any combination of files and folders that represent a type of data and can be validated as a group. As a starting point, we are focusing on single files, so "QC target" and "QC file" can be used interchangeably. However, we will quickly run into multi-file QC targets, such as FASTQ file pairs and BAM/CRAM files with their indices.
```

Comment:
An example of a "QC target" would be nice.  Just so I'm understanding, would an example of a "QC target" be "RNASeq FASTQ"?

Reply:
I added this:

> For example, RNA-seq FASTQ file pairs and BAM/CRAM files with their indices would be represented by a `QcFilePair`, whereas a more complex group of files would be represented by a `QcFileGroup`, which keeps tracks of relative paths.